IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m

```diff
@@ -215,4 +218,24 @@ - (MSIDPreferredAuthMethod)preferredAuthConfigurationEnumFromString:(NSString *)
     return MSIDPreferredAuthMethodNotConfigured;
 }
 
+- (NSString *)ssoProviderTypeStringFromEnum:(MSIDSsoProviderType)deviceMode
+{
+    switch (deviceMode) {
+        case MSIDCompanyPortalSsoProvider:
+            return @"companyPortal";
+        case MSIDMacBrokerSsoProvider:
+            return @"macBroker";
+        default:
+            return @"unknown";
+    }
+}
+
+- (MSIDSsoProviderType)ssoProviderTypeEnumFromString:(NSString *)deviceModeString
+{
+    if ([deviceModeString isEqualToString:@"companyPortal"])    return MSIDCompanyPortalSsoProvider;
```

Comment:
Create a dictionary <string, MSIDSsoProviderType> instead of if statements and return value for key. Return unknown by default in no matching key

Reply:
Hi @ameyapat  Cannot put `MSIDSsoProviderType` into dictionary as NSInteger is not supported neither as a key or value. I can convert `MSIDSsoProviderType` value to an NSNumber and save it into a dict, let me know.

Reply:
I meant to put this as a nit comment